### DISCUSSION TITLE
How to get different colors/themes working

### DISCUSSION THREAD

**Discussion by sebshader** [sebshader/pdnext#13] [General]
the easiest way to use themes these days is to install https://github.com/sebshader/color-themes-plugin, a tcl plugin that adds a preference dialog for choosing color themes

in order to make a theme: copy doc/7.stuff/colors-plugin.txt and change the extension to .tcl. 
Then change the values below "array set ::pd_colors {" using values from a hex color picker, or tk color names.

If you don't use the theme-picker plugin then you can put the file in your pd path to load it as a theme when pd-next starts.
Otherwise, put your themes into the "themes" folder in the color-themes-plugin folder (with a unique name ending in "-plugin.tcl"), and it will be loaded into the list in the preference window.

**Comment by Jamslerr** [comment]
Has this been tested on windows? I am having no luck getting the themes tcl plug in to work. Nothing shows up in the dialog box, and nothing shows up under the preferences menu.

**Reply by Jamslerr** [reply]
UNHANDLED ERROR: invalid command name "<!DOCTYPE"
    while executing
"<!DOCTYPE html>"
    ("uplevel" body line 7)
    invoked from within
"uplevel #0 $tclcode"
FAILED TO LOAD

I have created a folder titled "color-themes-plugin" folder and have inserted the tcl file into it as it states.

**Reply by sebshader** [reply]
@Jamslerr I can't remember if I've tested the plugin on windows but I would assume it works.. if not I can try to test it in a VM today or tomorrow maybe..

 so there are 2 separate ways to do themes: using the theme preference plugin, or using your own theme directly.

If you use the theme preference plugin (color-themes-plugin) method it should work to download & extract that folder (from the repo in the link above) and move the entire folder into your pd path, and make sure it's named "color-themes-plugin". You don't have to copy any .tcl or .txt files using that method. (and if you use color-themes-plugin you shouldn't have any other files in your pd path that set colors)

after you have it installed and working then you can add your own themes (.tcl files) to the "themes" folder inside that folder

if you've done all of that correctly then it might be some kind of encoding/doc type issue but since .tcl plugins have successfully been used cross-platform I don't think that's likely the case.

edit: using a hex editor, I can confirm there is no "<!DOCTYPE" in my .tcl files so something is also happening during some copying/downloading process as well.

**Comment by Jamslerr** [comment]
Am I misunderstanding the file path directory? Ive put it into the main Puredatanext folder. Also have tried adding it to startup, extra, doc, etc. there’s some times in my life where I make things extremely more complicated than necessary. It just keeps telling me it fails to load.

**Reply by sebshader** [reply]
according to http://msp.ucsd.edu/Pd_documentation/x4.htm it should work to put it in C:\Users\user_name\AppData\Roaming\Pd (where user_name is your user name)

**Reply by sebshader** [reply]
so the final path should be: C:\Users\user_name\AppData\Roaming\Pd\color-themes-plugin.

Inside that folder there should be the contents of the repo: color-themes-plugin.tcl, README.md, and the "themes" folder. When you download the repo do so by clicking on the green "Code" button, then "download zip". Then extract the folder and move it to that location

**Reply by sebshader** [reply]
also be aware that you have to restart pd-next in order to reload the tcl plugins after moving them to the correct paths

**Comment by Jamslerr** [comment]
Hey I got it finally. Thank you so much for the help.

**Reply by sebshader** [reply]
great, glad to hear it works on windows!

**Comment by Jamslerr** [comment]
Yeah, I downloaded the entire zip, extracted it into PD next, took the “main” off the end of the folder name, and it worked. So great, thank you.
